R/write_data.R

```diff
@@ -70,3 +70,35 @@ write_rds <- function(data, path, compress = "xz", ...) {
 
   return(invisible(data))
 }
+
+#' Write an R parquet file
+#'
+#' Wrapper around [arrow::write_parquet()], but with 'zstd' compression as
+#' default.
+#'
+#' @param data R object to write to serialise.
+#' @param path Path to where the data will be written.
+#' @param compress Compression method to use:
+#'     "uncompressed", "snappy", "gzip", "brotli", "zstd", "lz4", "lzo" or "bz2".
+#' @param ... Additional arguments to [write_parquet()][arrow::write_parquet()]
+#' for more details.
+#' @return `write_parquet()` returns the input `data` invisibly.
+#' @export
+#'
+#' @family write out data
+write_parquet <- function(data, path, compression = "zstd", ...) {
+  arrow::write_parquet(
+    x = data,
+    sink = path,
+    compression = compression,
+    version = "2.4",
```

Comment:
I think we should use `"latest"` here, or not include it at all, so it just uses `arrow::write_parquet()`'s default. By hardcoding the default to be anything else, there's a danger we are needlessly using an older version of the file specification which could eventually be deprecated.
```suggestion
    version = "latest",
```